src/main/java/games/strategy/triplea/ai/proAI/util/ProPurchaseUtils.java

```diff
@@ -190,6 +196,35 @@ public static boolean canUnitsBePlaced(final List<Unit> units, final PlayerID pl
     return placeUnits;
   }
 
+  public static Map<Territory, ProPurchaseTerritory> findBidTerritories(final PlayerID player) {
+
+    ProLogger.info("Find all bid territories");
+    final GameData data = ProData.getData();
+
+    // Find all territories that I can place units on
+    final Set<Territory> ownedOrHasUnitTerritories =
+        new HashSet<>(data.getMap().getTerritoriesOwnedBy(player));
+    ownedOrHasUnitTerritories.addAll(ProData.myUnitTerritories);
+    final List<Territory> potentialTerritories = Match.getMatches(ownedOrHasUnitTerritories,
+        Matches.TerritoryIsPassableAndNotRestrictedAndOkByRelationships(player, data, false, false, false, false,
+            false));
+
+    // Create purchase territory holder for each factory territory
+    final Map<Territory, ProPurchaseTerritory> purchaseTerritories = new HashMap<>();
+    for (final Territory t : potentialTerritories) {
+      final ProPurchaseTerritory ppt = new ProPurchaseTerritory(t, data, player, 1, true);
+      purchaseTerritories.put(t, ppt);
+      ProLogger.debug(ppt.toString());
+    }
+    return purchaseTerritories;
+  }
+
+  public static void incrementBidTerritories(final Map<Territory, ProPurchaseTerritory> purchaseTerritories) {
```

Comment:
"Missing a Javadoc comment."

Alternate fix is to minimize line count by using a lambda:

```java
purchaseTerritories.values().forEach(ppt -> ppt.setUnitProduction(ppt.getUnitProduction() + 1));
```

Until I actually wrote that code, I didn't realize that it was unit production being incremented.  So maybe a Javadoc would help anyway, or possibly renaming the method to include `UnitProduction` in the name?